modules/thermal_hydraulics/src/components/HeatTransferBase.C

```diff
@@ -139,7 +139,8 @@ HeatTransferBase::addHeatedPerimeter()
       makeFunctionControllableIfConstant(_P_hf_fn_name, "P_hf");
     }
 
-    getTHMProblem().addFunctionIC(_P_hf_name, _P_hf_fn_name, _flow_channel_subdomains);
+    if (!_app.isRestarting())
+      getTHMProblem().addFunctionIC(_P_hf_name, _P_hf_fn_name, _flow_channel_subdomains);
```

Comment:
_P_hf_fn_name is a user parameter right?
I feel we should have a warning if restarting and keeping the IC in the input file

Reply:
The function parameter is not just used for the IC but also the aux. Potentially a user could specify a time-dependent function.